pkg/api/graph.go

```diff
@@ -560,3 +560,25 @@ func LinkForImage(imageStream, tag string) StepLink {
 		return nil
 	}
 }
+
+// ClusterClaimLink determines what dependent link is required
+// for a job to claim a cluster
+func ClusterClaimLink() StepLink {
+	return &clusterClaimLink{}
+}
+
+type clusterClaimLink struct {
+}
+
+func (_ *clusterClaimLink) UnsatisfiableError() string {
+	return ""
+}
+
+func (_ *clusterClaimLink) SatisfiedBy(other StepLink) bool {
+	switch other.(type) {
+	case *clusterClaimLink:
+		return true
```

Comment:
Can we add a unique identifier to the step (test name should be enough, I think) and check it here, similar to the `internalImageStreamTagLink` linked above?

This implementation makes every `clusterClaimLink` satisfy every other, even if they are a result of different claims.  This is a bit pedantic as we don't currently generate jobs that would make this situation possible, but making the implementation robust is worth it here, IMO.